manticore/ethereum.py

```diff
@@ -458,25 +464,27 @@ def __getattribute__(self, name):
                 contract_account.add(1000)
          
         '''
-        if not name.startswith('_') and name in self._hashes.keys():
-            def f(*args, **kwargs):
-                caller = kwargs.get('caller', None)
-                value = kwargs.get('value', 0)
-                tx_data = ABI.make_function_call(str(self._hashes[name][0]), *args)
-                if caller is not None:
-                    caller = int(caller)
-                else:
-                    caller = self._default_caller
-                self._seth.transaction(caller=caller,
-                                        address=self._address,
-                                        value=value,
-                                        data=tx_data
-                                     )
-                self._caller = None
-                self._value = 0
-            return f
-        else:
-            return object.__getattribute__(self, name)            
+        if not name.startswith('_'):
+            self._init_hashes()
```

Comment:
won't this call always be a nop since, the ctor will execute `self._init_hashes()` before this?

Reply:
The ctor will execute this before indeed. But _init_hashes may not had the expected effect in that first call.

Reply:
hm ok, this is part of the context i lack. can you elaborate?

Reply:
I don't know if it helps, but this pull request fix https://github.com/trailofbits/manticore/issues/674

Reply:
oh definitely, that should be noted in the pr description